shimmingtoolbox/read_nii.py

```diff
@@ -0,0 +1,84 @@
+import json
+import math
+import nibabel as nib
+import os
+
+
+def read_nii(niiPath):
+    """
+
+    Args:
+        niiPath (str): Path to the nifti file
+
+    Returns:
+        image:
+        info:
+        jsonData:
+    """
+
+    info = nib.load(niiPath)
+    image = info.get_fdata()
+
+    # `extractBefore` should get the correct filename in both.nii and.nii.gz cases
+    jsonPath = niiPath.split('.nii')[0] + '.json'
```

Comment:
What happens if `niiPath = "we/love.nii/to/party.nii.gz.tmp/golly.nii"`?

Reply:
> What happens if `niiPath = "we/love.nii/to/party.nii.gz.tmp/golly.nii"`?

That would be really bad luck 😅 
Are you aware of any way to handle this ?

Reply:
@gaspardcereza yep. You use `dirname, fname = os.path.split(niiPath); jsonfname = fname.rsplit('.nii',1)[0] + '.json'; jsonPath = os.path.join(dirname, jsonfname)`.

Reply:
But it's gone away now.